lib/iris/fileformats/grib/_message.py

```diff
@@ -110,6 +108,8 @@ def data(self):
                 raise TranslationError(msg)
             if template == 90:
                 shape = (grid_section['Ny'], grid_section['Nx'])
+            elif template == 40 and reduced:
+                shape = (grid_section['numberOfDataPoints'],)
```

Comment:
OK fair enough for `_message`, but I think my point still stands for `grid_definition_section()`.